client/domain/SnapshotObjectFactory.js

```diff
@@ -0,0 +1,53 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating a snapshot of a student's code
+ * and performance at a given time.
+ */
+
+tie.factory('SnapshotObjectFactory', [
+  function() {
+    // Add support for a TieFeedback object.
```

Comment:
Comment should have TODO(eyurko): 

But I don't understand it. It looks like we do have a feedbackObject here already?